qiskit/visualization/pulse/matplotlib.py

```diff
@@ -601,6 +610,11 @@ def draw(self, schedule, dt, interp_method, plot_range,
             scale = scaling
         figure = plt.figure()
 
+        if not isinstance(scale, dict):
+            scale_dict_tmp = {chan: scale for chan in schedule.channels}
```

Comment:
What about just letting this be `scale`? I think `scale_dict` is also fine, but it doesn't feel very much like a `tmp`